src/transformers/models/ernie_m/tokenization_ernie_m.py

```diff
@@ -0,0 +1,354 @@
+# coding=utf-8
+# Copyright 2023 Xuan Ouyang, Shuohuan Wang, Chao Pang, Yu Sun, Hao Tian, Hua Wu, Haifeng Wang and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tokenization classes for ErnieM."""
+
+# Some of the code is Copied from original paddlenlp repository.(https://github.com/PaddlePaddle/PaddleNLP/blob/develop/paddlenlp/transformers/ernie_m/tokenizer.py)
+# and Albert directory.(https://github.com/huggingface/transformers/blob/main/src/transformers/models/albert/tokenization_albert.py)
+import io
+import unicodedata
+from typing import List, Optional
+
+import sentencepiece as spm
+
+from ...tokenization_utils import PreTrainedTokenizer
+
+
+__all__ = ["ErnieMTokenizer"]
+
+SPIECE_UNDERLINE = "▁"
+VOCAB_FILES_NAMES = {"vocab_file": "vocab.txt", "sentencepiece_model_ckpt": "sentencepiece.bpe.model"}
+
+
+class ErnieMTokenizer(PreTrainedTokenizer):
+    r"""
+    Args:
+    Constructs a ErnieM tokenizer. It uses the `sentencepiece` tools to cut the words to sub-words.
+        vocab_file (str):
+            The file path of the vocabulary.
+        sentencepiece_model_file (str):
+            The file path of sentencepiece model.
+        do_lower_case (str, optional):
+            Whether or not to lowercase the input when tokenizing. Defaults to`True`.
+        unk_token (str, optional):
+            A special token representing the *unknown (out-of-vocabulary)* token. An unknown token is set to be
+            `unk_token` inorder to be converted to an ID. Defaults to "[UNK]".
+        sep_token (str, optional):
+            A special token separating two different sentences in the same input. Defaults to "[SEP]".
+        pad_token (str, optional):
+            A special token used to make arrays of tokens the same size for batching purposes. Defaults to "[PAD]".
+        cls_token (str, optional):
+            A special token used for sequence classification. It is the last token of the sequence when built with
+            special tokens. Defaults to "[CLS]".
+        mask_token (str, optional):
+            A special token representing a masked token. This is the token used in the masked language modeling task
```

Comment:
Please respect our docstring guidelines: type should be in code style, optional in italics, the default should be included in the line of the arg.